### PR TITLE
test: add unit tests for logger class [DET-7901]

### PR DESCRIPTION
**Pull request by hamidzr**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/Makefile [hunk 1]
```diff
@@ -88,4 +88,4 @@ push-shared: test-shared
 
 	# fail if there are uncommitted changes.
 	git diff --exit-code || (echo "uncommitted changes, please commit or stash" ; false)
-	cd $(git rev-parse --show-toplevel) && git subtree push --prefix=webui/react/src/shared https://github.com/determined-ai/shared-web main
+	cd $(shell git rev-parse --show-toplevel) && git subtree push --prefix=webui/react/src/shared https://github.com/determined-ai/shared-web main
```
Comment (keita-determined) [inline webui/react/Makefile:91]: what's the different with and without `shell`?

**Review by keita-determined — APPROVED** [review]
LGMT

File: webui/react/Makefile (hunk 1 above)
Comment (hamidzr) [reply webui/react/Makefile:91]: shell run the arguments as a shell command. without it the contents get evaluted to "" not sure exactly how make evalutes the args. anyway this is supposed to go in through another pr https://github.com/determined-ai/determined/pull/4672 I'll take it out from here

**Comment by hamidzr** [comment]
ops. I thought I'm enabling auto merge (which lands the changes after and if all tests passed)

**Comment by hamidzr** [comment]
thanks for the review

**Comment by hamidzr** [comment]
fwiw all tests had passed the prior run https://app.circleci.com/pipelines/github/determined-ai/determined?branch=test-logger

**Pull request merged**